packages/graphql/src/translate/translate-top-level-cypher.ts

```diff
@@ -0,0 +1,225 @@
+/*
```

Comment:
I think this file's readability (and subsequent ease of collaboration) would benefit from a few things:

- Declaring variables only just before usage. For example, if a variable is used only once in an if statement, I think it would be easier to read the file in a general sense if they were placed together.
- Being broken up into smaller functions. For example the two blocks `if (referenceNode)` and `if (referenceUnion) {`

Reply:
I completely agree with you, but keep in mind that this is a block of code that was moved from the cypher resolver into a separate function (it is not new code, the diff is misleading). Refactor this further is a bit of a scope creep for this task, although I'm happy to implement some quick wins if you see them